http-netty/src/main/java/io/micronaut/http/netty/NettyFormUrlEncodedDecoder.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.netty;
+
+import io.micronaut.http.form.FormConfiguration;
+import io.micronaut.http.form.FormUrlEncodedDecoder;
+import jakarta.inject.Singleton;
+import io.netty.handler.codec.http.QueryStringDecoder;
+import reactor.util.annotation.NonNull;
+
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation of {@link FormUrlEncodedDecoder} based on Netty's {@link QueryStringDecoder}.
+ * @author Sergio del Amo
+ * @since 4.7.1
+ */
+@Singleton
+public class NettyFormUrlEncodedDecoder implements FormUrlEncodedDecoder {
```

Comment:
it is still unclear to me why we need custom code for Netty when we have `QueryStringDecoder` forked from Netty in the `http` module

Reply:
[I deleted the netty based implementation ](https://github.com/micronaut-projects/micronaut-core/pull/11272/commits/e3a9611787265b4a75f978d20d3e3977aa03a510).